### PR TITLE
LLT-5068 Bump Moose 4.0.1

### PR DESCRIPTION
**Pull request by matislovas**
Bumping Moose to v4.0.1

### :ballot_box_with_check: Definition of Done checklist
- [x] Commit history is clean ([requirements](../blob/main/docs/git_commit_messages_requirements.md))
- [x] README.md is updated
- [x] changelog.md is updated
- [x] Functionality is covered by unit or integration tests

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -21,7 +21,7 @@ cfg-if = "1.0.0"
 ffi_helpers = "0.3.0"
 num_cpus = "1.15.0"
 wasm-bindgen = "=0.2.83" # Only for compatability with moose wasm version
-h2 = "=0.3.24" # RUSTSEC-2024-0003
```
Comment (packgron) [inline Cargo.toml:24]: You probably need to rebase, fix is already in main

**Review by matszczygiel — COMMENTED** [review]
+1

File: Cargo.toml (hunk 1 above)
Comment (matislovas) [reply Cargo.toml]: Rebased

**Review by jjanowsk — APPROVED** [review]
+1

**Review by matszczygiel — COMMENTED** [review]
+1

**Pull request merged**